### DISCUSSION TITLE
Request: Update orders_totals handling to capture applied tax-rate

### DISCUSSION THREAD

**Discussion by lat9** [zencart/zencart#6122] [Ideas]
For v2 (and possibly a 1.5.7, if projected), I suggest that the `order_totals` processing be updated so that any taxable order-total modules capture the tax-rate applied to an order, with the `order` class saving those values in a new `order_totals::tax_rate` column.

The captured tax-rate will improve any after-order analysis and/or editing.

**Comment by drbyte** [comment]
Were you thinking of capturing only a single tax-rate? 
How would you handle products having varying tax rates/classes?

**Comment by zcwilt** [comment]
While we haven't posted much about things like order composition and things that affect order totals such as shipping/gvs etc in v2, it's always been known that Zen Cart has ongoing issues with rounding of tax values and while we have tried to update code over time to address this, tax calculation code is still incredibly fragile in this regard.

One way of addressing this is to move to a ``line item`` model for storing  order details. In this model every item that affects an order would store details of it's tax entries as well as other details. I will try and post something about this in the project board at some point, along with the benefits and cons of using ``line items``

**Comment by lat9** [comment]
> Were you thinking of capturing only a single tax-rate?
> How would you handle products having varying tax rates/classes?

@drbyte, my suggestion applies to the `order_totals` not the products.  

I was considering an interface through which an order-total could identify its tax rate (and tax) in a manner similar to the current shipping modules and products' information recording at order placement.

**Comment by scottcwilson** [comment]
@lat9 are you still keen on this?  If not, please close; if so, please add label "Implement in Next Release" (or "Consider in Next Release") as you deem appropriate.

**Comment by scottcwilson** [comment]
Moving to discussions.
